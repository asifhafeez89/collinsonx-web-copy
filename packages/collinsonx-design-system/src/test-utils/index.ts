import userEvent from '@testing-library/user-event';

export * from '@testing-library/react';
export { userEvent };
export { default as render, Provider } from './render';
