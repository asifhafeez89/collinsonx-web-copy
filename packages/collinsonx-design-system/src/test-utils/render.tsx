import { RenderOptions, render } from '@testing-library/react';
import { MantineProvider } from '@mantine/core';
import { ReactElement, ReactNode } from 'react';

export const Provider = ({ children }: { children: ReactNode }) => (
  <MantineProvider>{children}</MantineProvider>
);

type Render = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => ReturnType<typeof render>;

const customRender: Render = (ui, options) =>
  render(ui, {
    wrapper: Provider as unknown as RenderOptions['wrapper'],
    ...options,
  });

export default customRender;
