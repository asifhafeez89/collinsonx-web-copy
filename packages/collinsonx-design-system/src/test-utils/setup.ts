import userEvent from '@testing-library/user-event';
import render from './render';

export default function (jsx: Parameters<typeof render>[0]) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}
