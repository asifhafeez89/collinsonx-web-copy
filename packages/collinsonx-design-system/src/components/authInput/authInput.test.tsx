import renderer from 'react-test-renderer';
import AuthInput from '.';
import { render, userEvent, waitFor, Provider } from 'test-utils';

const mockFn = jest.fn();
describe('<AuthInput />', () => {
  it('renders AuthInput', () => {
    const tree = renderer
      .create(
        <Provider>
          <AuthInput handleCodeChange={mockFn} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  xit('can type ', async () => {
    const { container } = render(
      <>
        <AuthInput handleCodeChange={mockFn} />
      </>
    );

    const inputs = container.querySelectorAll('input');

    userEvent.type(inputs[0], '123456');

    await waitFor(() => {
      expect(mockFn).toHaveBeenCalledTimes(6);
      // expect(mockFn).toHaveBeenLastCalledWith('123456');
    });
  });
});
