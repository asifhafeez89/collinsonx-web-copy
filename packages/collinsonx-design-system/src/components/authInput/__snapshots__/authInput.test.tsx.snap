// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthInput /> renders AuthInput 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": "@media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}",
      }
    }
    data-mantine-styles="classes"
  />,
  .emotion-0 input {
  font-weight: 600;
  font-size: 18px;
  width: 48px;
  height: 48px;
  padding: 0;
  font-size: 1rem;
  text-align: center;
  margin-right: 8px;
  text-transform: uppercase;
  border-radius: 8px;
  color: #494949;
  border: 1px solid #d6d6d6;
  background: #fff;
  -webkit-background-clip: padding-box;
  background-clip: padding-box;
}

.emotion-0 input:last-child {
  margin-right: 0px;
}

<div
    className="emotion-0"
  >
    <div>
      <input
        aria-label="Character 1."
        autoComplete="one-time-code"
        inputMode="text"
        maxLength={1}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onPaste={[Function]}
        pattern="[a-zA-Z0-9]{1}"
        placeholder="-"
        type="text"
      />
      <input
        aria-label="Character 2."
        autoComplete="off"
        inputMode="text"
        maxLength={1}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onPaste={[Function]}
        pattern="[a-zA-Z0-9]{1}"
        placeholder="-"
        type="text"
      />
      <input
        aria-label="Character 3."
        autoComplete="off"
        inputMode="text"
        maxLength={1}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onPaste={[Function]}
        pattern="[a-zA-Z0-9]{1}"
        placeholder="-"
        type="text"
      />
      <input
        aria-label="Character 4."
        autoComplete="off"
        inputMode="text"
        maxLength={1}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onPaste={[Function]}
        pattern="[a-zA-Z0-9]{1}"
        placeholder="-"
        type="text"
      />
      <input
        aria-label="Character 5."
        autoComplete="off"
        inputMode="text"
        maxLength={1}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onPaste={[Function]}
        pattern="[a-zA-Z0-9]{1}"
        placeholder="-"
        type="text"
      />
      <input
        aria-label="Character 6."
        autoComplete="off"
        inputMode="text"
        maxLength={1}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onPaste={[Function]}
        pattern="[a-zA-Z0-9]{1}"
        placeholder="-"
        type="text"
      />
    </div>
  </div>,
]
`;
